src/CommonLib/LDAPUtils.cs

```diff
@@ -572,6 +572,38 @@ public IEnumerable<string> DoRangedRetrieval(string distinguishedName, string at
             }
         }
 
+        /// <summary>
+        /// Takes a host in most applicable forms from AD and attempts to resolve it into a SID, falling back to hostname if SID cannot be resolved.
+        /// </summary>
+        /// <param name="hostname"></param>
+        /// <param name="domain"></param>
+        /// <returns></returns>
+        public async Task<string> ResolveHostToSidWithHostnameFallback(string hostname, string domain)
```

Comment:
Here's the meat and potatoes.  Split ResolveHostToSid into two functions, one with and one without hostname resolution fallback.